File: solutions/csharp/hamming/2/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
            throw new ArgumentException("Arguments must both have the same length.");

        return Enumerable
            .Range(0, firstStrand.Length)
            .Count(i => firstStrand[i] != secondStrand[i]);
    }
}

File: solutions/go/hamming/2/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
    if len(a) != len(b) {
        return -1, errors.New("Arguments must be the same length.")
    }

    count := 0

    for i := range a {
        if a[i] != b[i] {
            count++
        }
    }

    return count, nil
}
